src/viking.js

```diff
@@ -1,11 +1,98 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    
+    constructor(health, strength){
+        this.health = health;
+        this.strength = strength;
+    
+    }
+    attack(){
+       return this.strength;
+    }
+
+    receiveDamage(damage){
+        this.health -= damage;
+    }   
+};
+  
+
+    
+  
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name,health,strength){
+        super(health,strength)
+        this.name = name;
+    }
+    receiveDamage(damage){
+       this.health -= damage;
+        if(this.health <=0){
+            return `${this.name} has died in act of combat`; 
+        }else{
+            return `${this.name} has received ${damage} points of damage`;
+        }
+    }
+    
+    battleCry(){
+        return "Odin Owns You All!";
+    }
+};
+    
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+
+    receiveDamage(damage){
+        this.health -= damage;
+        if(this.health <=0){
+            return `A Saxon has died in combat`; 
+        }else{
+            return `A Saxon has received ${damage} points of damage`;
+        }   
+    }
+};
 
 // War
-class War {}
+class War{
+    vikingArmy = [];
+    saxonArmy = [];
+
+    addViking(Viking){
+    this.vikingArmy.push(Viking);
+    }
+    addSaxon(Saxon){
+    this.saxonArmy.push(Saxon);
+    }
+    vikingAttack(){
+     let randomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+     let randomViking = Math.floor(Math.random()*this.vikingArmy.length);
+     this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
+
+     if(this.vikingArmy[randomViking].health <=0){
+        this.vikingArmy.splice(randomViking)     
```

Comment:
Aquí en el splice sería `this.vikingArmy.splice(randomViking, 1)` ya que tenemos que indicar que sólo queremos eliminar un elemento del array, de otra manera eliminaría todos los elementos que hay desde el indice indicado hasta el final